dozent/dozent.py

```diff
@@ -108,20 +138,40 @@ def download_timeframe(self, start_date: datetime.date, end_date: datetime.date,
 
         os.makedirs(download_dir, exist_ok=True)
 
-        for x in range(multiprocessing.cpu_count()):
-            worker = _DownloadWorker(queue, download_dir, verbose)
-            # worker.set_verbosity(verbose=verbosity)
-            # Setting daemon to True will let the main thread exit even though the workers are blocking
-            worker.daemon = True
-            worker.start()
+        number_of_dates = (end_date - start_date).days + 1
 
-        for sample_date in self.get_links_for_days(start_date=start_date, end_date=end_date):
-            print(f"Queueing tweets download for {sample_date['month']}-{sample_date['year']}")
-            queue.put(sample_date['link'])
+        task_id = 0
+
+        if number_of_dates < 4:
+            for x in range(multiprocessing.cpu_count() * 2):
+                worker = _DownloadWorker(
+                    queue, download_dir, task_id, number_of_dates, verbose
+                )
+                # Setting daemon to True will let the main thread exit even though the workers are blocking
+                worker.daemon = True
+                worker.start()
+        else:
+            for x in range(number_of_dates):
+                worker = _DownloadWorker(
+                    queue, download_dir, task_id, number_of_dates, verbose
+                )
+                # Setting daemon to True will let the main thread exit even though the workers are blocking
+                worker.daemon = True
+                worker.start()
+
+        for sample_date in self.get_links_for_days(
+            start_date=start_date, end_date=end_date
+        ):
+            print(
+                f"Queueing tweets download for {sample_date['day']}-{sample_date['month']}-{sample_date['year']}"
+            )
+            queue.put(sample_date["link"])
```

Comment:
Consider breaking this down into separate functions? Might make it easier to understand what's going on